TeamCode/src/main/java/org/firstinspires/ftc/teamcode/PixyCam.java

```diff
@@ -0,0 +1,170 @@
+package org.firstinspires.ftc.teamcode;
+
+import android.util.Log;
+
+import com.qualcomm.robotcore.hardware.I2cDeviceSynch;
+
+import java.util.Arrays;
+import java.util.Collections;
+
+/**
+ * Created by NoahR on 10/3/17.
+ * Note about Modern Robotics I2C wiring. (Left to right)
+ * 0 - 1 - 2 - 3
+ * 5v - Data - Clock - gnd
+ */
+
+public class PixyCam
+{
+    I2cDeviceSynch m_deviceClient;
+    private byte[] m_registerValues;
+    private int m_dataStartAddress = 0;
+
+    private final int START_ADDRESS = 0x54;
+    private final int READ_LENGTH = 26; // Start read at 0x58, end at 0x58 + 26
+    private final byte[] PIXY_DEVICE_SIGNATURE = {85, (byte) 170, 85, (byte) 170};
+    private final int SIGNATURE_SPACE_SIZE = 10; // Device signature will be in the first 10 bytes
+
```

Comment:
Default constructor? I'd avoid it, don't let people create malformed objects whenever possible! This code could take a long time to debug when mixed in with other things:

    PixyCam mycam = new PixyCam();
    myCam.Update();

Additionally, consider adding extra checks and throwing exceptions/errors if *anything* is off; failing earlier is better. For example, can you probe the `deviceClient` with `getDeviceName()` and/or `getManufacturer()` to make sure the vendor string matches? What if somebody accidentally passed you the wrong i2c device/address? You could save a teammate a lot of debugging time here.